Add a search field on model selector / improve mobile display

ServeurpersoCom opened the pull request:
*Make sure to read the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*

Add a search field to the model selector / Improve mobile display / Simplify the code

<img width="848" height="564" alt="PC1" src="https://github.com/user-attachments/assets/cdd1aac6-4084-40a0-a46d-860a381e542e" />
<img width="803" height="543" alt="PC2" src="https://github.com/user-attachments/assets/483683e8-0af6-42cb-9351-ad201d5d6055" />

<img width="320" height="693" alt="Screenshot_20251206_140505_Kiwi Browser" src="https://github.com/user-attachments/assets/24315909-e89d-4755-92fc-5f83da6fde80" />
<img width="320" height="693" alt="Screenshot_20251206_140527_Kiwi Browser" src="https://github.com/user-attachments/assets/5c300597-13d1-44da-9970-b5708524b305" />

Close #17741

ServeurpersoCom committed:
webui: add search field to model selector and fixes mobile viewport overflow

ServeurpersoCom committed:
webui: simplify model search style and code

allozaur committed:
refacor: Search Input component & consistent UI for Models Selector search

allozaur committed:
feat: Use Popover component + improve interactions

allozaur reviewed (changes requested):
Added few improvements in ServeurpersoCom/llama.cpp/pull/3 ;)

allozaur committed:
fix: Fetching props for only loaded models in ROUTER mode

ServeurpersoCom committed:
webui: prevent models selector popover from overflowing viewport

Use Floating UI's auto-positioning with 50dvh height limit and proper
collision detection instead of forcing top positioning. Fixes overflow
on desktop and mobile keyboard issues

ServeurpersoCom commented:
Now we're good @allozaur ! I adjusted the popover positioning to prevent it from overflowing the screen on both desktop and mobile. The main changes are removing the forced side="top" positioning to let Floating UI handle placement automatically, and using max-h-[50dvh] to ensure the menu adapts to the actual viewport size (including when the mobile keyboard is open). I also restructured the layout with proper flexbox so scrolling works correctly

allozaur reviewed (approved):
Let's just update the image/video demos in the description and we are good to go :)

ServeurpersoCom commented:
If we are demanding in terms of UX ergonomics the search field could be placed below the list to minimize user input movement when the list is compressed during a search. (the popover can already "flip" above or below depending on the available space)

ServeurpersoCom committed:
webui: keep search field near trigger in models selector

Place search at the 'near end' (closest to trigger) by swapping layout
with CSS flexbox order based on popover direction. Prevents input from
moving during typing as list shrinks

ServeurpersoCom committed:
chore: update webui build output
